chain/network/src/peer_manager/tests.rs

```diff
@@ -156,6 +166,93 @@ async fn no_edge_broadcast_after_restart() {
     }
 }
 
+// Nonces must be odd (as even ones are reserved for tombstones).
+fn to_odd_nonce(value: u64) -> u64 {
+    if value % 2 == 0 {
+        value + 1
+    } else {
+        value
+    }
+}
+
+// Test connecting to peer manager with timestamp-like nonces.
+#[tokio::test]
+async fn test_nonces() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    //let mut total_edges = vec![];
+    let store = create_test_store();
+
+    // Start a PeerManager and connect a peer to it.
+    let pm = peer_manager::testonly::start(
+        store.clone(),
+        chain.make_config(rng),
+        chain.clone(),
+        clock.clock(),
+    )
+    .await;
+    // Try to connect with peer with a valid nonce (current timestamp).
+    {
+        let cfg = peer::testonly::PeerConfig {
+            signer: data::make_signer(rng),
+            chain: chain.clone(),
+            peers: vec![],
+            start_handshake_with: Some(PeerId::new(pm.cfg.node_key.public_key())),
+            force_encoding: Some(Encoding::Proto),
+            // Connect with nonce equal to unix timestamp
+            nonce: Some(to_odd_nonce(clock.now_utc().unix_timestamp() as u64)),
+        };
+        let stream = TcpStream::connect(pm.cfg.node_addr.unwrap()).await.unwrap();
+        let mut peer =
+            peer::testonly::PeerHandle::start_endpoint(clock.clock(), rng, cfg, stream).await;
+        peer.complete_handshake().await;
+    }
+
+    // Now try the peer with invalid timestamp (in the past)
+    {
+        let cfg = peer::testonly::PeerConfig {
+            signer: data::make_signer(rng),
+            chain: chain.clone(),
+            peers: vec![],
+            start_handshake_with: Some(PeerId::new(pm.cfg.node_key.public_key())),
+            force_encoding: Some(Encoding::Proto),
+            // Connect with nonce equal to unix timestamp
+            nonce: Some(to_odd_nonce(
+                clock.now_utc().checked_sub(time::Duration::days(1)).unwrap().unix_timestamp()
```

Comment:
clock.now_utc().checked_sub(duration).unwrap() is equivent to just (clock.now_utc() - duration)

Reply:
done.